packages/core/frontend/sirius-components-core/src/workbench/Workbench.tsx

```diff
@@ -63,180 +68,233 @@ const isRepresentationRenamedEventPayload = (
+    };
 
-  const workbenchViewLeftSideContributions: WorkbenchViewContribution[] = [];
-  const workbenchViewRightSideContributions: WorkbenchViewContribution[] = [];
+    useEffect(() => {
+      if (
+        state.displayedRepresentationMetadata &&
+        state.displayedRepresentationMetadata.id !== initialRepresentationSelected?.id
+      ) {
+        onRepresentationSelected(state.displayedRepresentationMetadata);
+      } else if (state.displayedRepresentationMetadata === null && initialRepresentationSelected) {
+        onRepresentationSelected(null);
+      }
+    }, [onRepresentationSelected, initialRepresentationSelected, state.displayedRepresentationMetadata]);
+
+    const workbenchViewLeftSideContributions: WorkbenchViewContribution[] = [];
+    const workbenchViewRightSideContributions: WorkbenchViewContribution[] = [];
 
-  const { data: workbenchViewContributions } = useData(workbenchViewContributionExtensionPoint);
-  for (const workbenchViewContribution of workbenchViewContributions) {
-    if (workbenchViewContribution.side === 'left') {
-      workbenchViewLeftSideContributions.push(workbenchViewContribution);
-    } else if (workbenchViewContribution.side === 'right') {
-      workbenchViewRightSideContributions.push(workbenchViewContribution);
+    const { data: workbenchViewContributions } = useData(workbenchViewContributionExtensionPoint);
+    for (const workbenchViewContribution of workbenchViewContributions) {
+      if (workbenchViewContribution.side === 'left') {
+        workbenchViewLeftSideContributions.push(workbenchViewContribution);
+      } else if (workbenchViewContribution.side === 'right') {
+        workbenchViewRightSideContributions.push(workbenchViewContribution);
+      }
     }
-  }
 
-  const { Component: MainComponent } = useComponent(workbenchMainAreaExtensionPoint);
-  let main = <MainComponent editingContextId={editingContextId} readOnly={readOnly} />;
+    const leftPanelConfiguration: WorkbenchSidePanelConfiguration | null =
+      initialWorkbenchConfiguration?.sidePanels?.find(
+        (configuration) => configuration && configuration.id === 'left'
+      ) ?? null;
+    const rightPanelConfiguration: WorkbenchSidePanelConfiguration | null =
+      initialWorkbenchConfiguration?.sidePanels?.find(
+        (configuration) => configuration && configuration.id === 'right'
+      ) ?? null;
+    const mainAreaConfiguration: WorkbenchMainPanelConfiguration | null =
+      initialWorkbenchConfiguration?.mainPanel ?? null;
 
-  const onRepresentationClick = (representation: RepresentationMetadata) => {
-    setSelection({ entries: [{ id: representation.id }] });
-  };
+    const { Component: MainComponent } = useComponent(workbenchMainAreaExtensionPoint);
+    let main = (
+      <MainComponent
+        editingContextId={editingContextId}
+        readOnly={readOnly}
+        initialConfiguration={mainAreaConfiguration}
```

Comment:
I don't understand how this area and `RepresentationNavigation` could share the same configuration, it doesn't make sense to change anything in `MainComponent`